src/app/hooks.ts

```diff
@@ -0,0 +1,6 @@
+import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from './store';
+
+export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
+
+export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
```

Comment:
The `useDispatch.withTypes<AppDispatch>()` syntax is incorrect. Instead, you should use `useDispatch<AppDispatch>()` to correctly type the `useDispatch` hook with `AppDispatch`.